dali/operators/imgcodec/peek_image_shape.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/imgcodec/peek_image_shape.h"
+#include "dali/operators/imgcodec/operator_utils.h"
+#include "dali/imgcodec/registry.h"
+
+namespace dali {
+namespace imgcodec {
+
+DALI_SCHEMA(experimental__PeekImageShape)
+  .DocStr(R"code(Obtains the shape of the encoded image.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .AddOptionalTypeArg("dtype",
+    R"code(Data type, to which the sizes are converted.)code", DALI_INT64);
```

Comment:
We should add two more arguments:
- output_type (DALIImageFormat)
- adjust_orientation (bool)
The latter should be added to the decoder as well.

Reply:
Added